src/useLifeCycleInterfaceRule.ts

```diff
@@ -61,28 +61,21 @@ export class ClassMetadataWalker extends Lint.RuleWalker {
     return interfaces;
   }
 
-  private validateMethods(methods: any[], interfaces: string[], className: string) {
+  private validateMethods(methods: ts.ClassElement[], interfaces: string[], className: string) {
     methods.forEach(m => {
-      let n = (<any>m.name).text;
-      if (n && this.isMethodValidHook(m, interfaces)) {
-        let hookName = n.substr(2, n.lenght);
```

Comment:
Fun fact: `lengHt` was mispelled (2 times in this class) :)

Reply:
Oh!